sys-boot/lilo/lilo-24.2-r3.ebuild

```
@@ -0,0 +1,197 @@
+}
+
+src_compile() {
+	# lilo needs this. bug #140209
+	export LC_ALL=C
+
+	# we explicitly prevent the custom CFLAGS for stability reasons
+	if use static; then
+		local target=alles
+	else
+		local target=all
+	fi
+
+	emake CC="$(tc-getCC) ${LDFLAGS}" ${target}
+}
+
+src_install() {
+	emake DESTDIR="${ED}" install
+
+	if use !minimal; then
+		into /
+		dosbin "${WORKDIR}"/dolilo/dolilo
+
+		into /usr
+		dosbin keytab-lilo.pl
+
+		insinto /etc
+		newins "${FILESDIR}"/lilo.conf lilo.conf.example
+
+		newconfd "${WORKDIR}"/dolilo/dolilo.conf.d dolilo.example
+
+		dodoc CHANGELOG* readme/README.* readme/INCOMPAT README
+		docinto samples ; dodoc sample/*
+	fi
+
+	# This we don't use
+	rm -r "${ED}/etc/initramfs" || die
+	# This should be in /usr/lib and it should have the .install suffix
+	dodir /usr/lib/kernel
+	for dir in postinst.d postrm.d; do
+		mv "${ED}/etc/kernel/${dir}" "${ED}/usr/lib/kernel/${dir}" || die
+		mv "${ED}/usr/lib/kernel/${dir}/zz-runlilo" "${ED}/usr/lib/kernel/${dir}/90-runlilo.install" || die
+	done
+	# Insert wrapper for systemd's kernel-install
+	exeinto /usr/lib/kernel/install.d
+	newexe - 90-runlilo.install <<-EOF
+		#!/bin/sh
+		if [ "\${1}" = "add" ]; then
+			exec "\$(dirname \${0})/../postinst.d/90-runlilo.install"
+		elif [ "\${1}" = "remove" ]; then
+			exec "\$(dirname \${0})/../postrm.d/90-runlilo.install"
+		fi
+	EOF
+}
+
+# Check whether LILO is installed
+# This function is from /usr/sbin/mkboot from debianutils, with copyright:
+#
+#   Debian GNU/Linux
+#   Copyright 1996-1997 Guy Maor <maor@debian.org>
+#
+# Modified for Gentoo for use with the lilo ebuild by:
+#   Martin Schlemmer <azarah@gentoo.org> (16 Mar 2003)
+#
+lilocheck() {
+	local bootpart=
+	local rootpart="$(mount | grep -v "tmpfs" | grep -v "rootfs" | grep "on / " | cut -f1 -d " ")"
+
+	echo
+	ebegin "Checking whether LILO can be safely updated"
+
+	if [ "$(whoami)" != "root" ]; then
+		eend 1
+		eerror "Only root can check for LILO"
+		return 1
+	fi
+
+	if [ -z "${rootpart}" ]; then
+		eend 1
+		eerror "Could not determine root partition"
+		return 1
+	fi
+
+	if [ ! -f /etc/lilo.conf -o ! -x /sbin/lilo ]; then
+		eend 1
+		eerror "No LILO configuration in place"
+		return 1
+	fi
+
+	bootpart="$(sed -n "s:^boot[ ]*=[ ]*\(.*\)[ ]*:\1:p" /etc/lilo.conf)"
+
+	if [ -z "${bootpart}" ]; then
+		# lilo defaults to current root when 'boot=' is not present
+		bootpart="${rootpart}"
+	fi
+
+	if ! dd if=${bootpart} ibs=16 count=1 2>&- | grep -q 'LILO'; then
+		eend 1
+		eerror "No LILO signature on ${bootpart}"
+		ewarn "Check your /etc/lilo.conf, or run /sbin/lilo by hand."
+		return 1
+	fi
+
+	if grep -q "^[[:space:]]*password[[:space:]]*=[[:space:]]*\"\"" /etc/lilo.conf; then
+		eend 1
+		eerror "Interactive password entry configured"
+		ewarn "Run /sbin/lilo -p by hand."
+		return 1
+	fi
+
+	einfo "LILO on ${bootpart} is safe to update"
+	eend 0
+	return 0
+}
+
+pkg_postinst() {
+	if [ ! -e "${ROOT}/boot/boot.b" -a ! -L "${ROOT}/boot/boot.b" ]; then
```

Can you change this to Bash tests? `-a` is also obsolescent and IIRC has some pitfalls but I don't remember what they are, i.e.

```suggestion
	if [[ ! -e "${ROOT}/boot/boot.b" && ! -L "${ROOT}/boot/boot.b" ]]; then
```

and then similar elsewhere.

---

> The -a and -o binary primaries and the '(' and ')' operators have been removed. (Many expressions using them were ambiguously defined by the grammar depending on the specific expressions being evaluated.) Scripts using these expressions should be converted to the forms given below. Even though many implementations will continue to support these forms, scripts should be extremely careful when dealing with user-supplied input that could be confused with these and other primaries and operators. Unless the application developer knows all the cases that produce input to the script, invocations like:
>
> test "$1" -a "$2"
> 
> should be written as:
> 
> test "$1" && test "$2"
> 
> to avoid problems if a user supplied values such as $1 set to '!' and $2 set to the null string. That is, replace: